app/viewModels/pagination/PaginationViewModel.scala

```diff
@@ -64,13 +65,31 @@ object PaginationViewModel {
     }
 
     val previous: Option[PaginationLink] = if (currentPage > 1) {
-      Some(PaginationLink(hrefWithParams(currentPage - 1)))
+      Some(
+        PaginationLink(
+          hrefWithParams(currentPage - 1),
+          attributes = navigationHiddenText
+            .map(
+              text => Map("aria-label" -> s"""${messages("govuk.pagination.previous")} page of ${text.toLowerCase}""")
+            )
+            .getOrElse(Map.empty)
+        )
+      )
     } else {
       None
     }
 
     val next: Option[PaginationLink] = if (currentPage < results.totalPages) {
-      Some(PaginationLink(hrefWithParams(currentPage + 1)))
+      Some(
+        PaginationLink(
+          href = hrefWithParams(currentPage + 1),
+          attributes = navigationHiddenText
+            .map(
+              text => Map("aria-label" -> s"""${messages("govuk.pagination.next")} page of ${text.toLowerCase}""")
```

Comment:
This is really good! Very minor but can we have two message keys for the previous and next i.e.
```
next.hidden = Next page of {0}
previous.hidden = Previous page of {0}
```
for example.